### PR TITLE
Add devise for user sign up

### PR DESCRIPTION
**Pull request by Berkay-akbas**
# Hello Code Reviewer 👋 

# Changes ♻️ 
- You can register a new user with devise now.
- User logs in with email and password.
- Stored passwords are hashed in the DB.
- Asks for a confirmation email but confirmation required disabled within the user model by calling devise helper method for testing purposes.
- Password is resettable.

Thanks for reviewing our code!

### CODE REVIEW

File: app/views/devise/registrations/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    <%= f.label :email %><br />
+    <%= f.email_field :email, autofocus: true, autocomplete: "email" %>
+  </div>
+
+
+  <div class="field">
+    <%= f.label :name %><br />
+    <%= f.text_field :name %>
+  </div>
+
```
Comment (julie-ify) [inline app/views/devise/registrations/new.html.erb:16]: - [ ] I am unable to sign up as a user; when I click the button nothing happens. Your devise error partial does not display anything so I had to log the errors manually. Your validation for name returns errors. Kindly look into that and fix it.

  chy

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @Berkay-akbas and @developerwaleed,

Good job so far!

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/devise/registrations/new.html.erb (hunk 1 above)
Comment (Berkay-akbas) [reply app/views/devise/registrations/new.html.erb:16]: @julie-ify  Looks like the missing part was before_action callback for the ApplicationController. It is now fixed. Thanks for the review!

**Review by FrederickMih — APPROVED** [review]
Hi @Berkay-akbas , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights :green_circle: 

- PR is descriptive :heavy_check_mark: 
- No Linter error :heavy_check_mark: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**